Source/Redemption/Private/PowerUpComponent.cpp

```diff
@@ -83,4 +87,24 @@ void UPowerUpComponent::ChangeActivePowerUp(EActivePowerUp NewPower)
 {
 	CurrentPowerUp = NewPower;
 	ChangeMappingContext();
+}
+
+//will spawn the bone
+void UPowerUpComponent::ExecuteBonePowerUp()
+{
+	if (BoneBP)
+	{
+		if (ACharacter* OwnerChar = Cast<ACharacter>(GetOwner()))
+		{
+			FVector FowardVec = OwnerChar->GetActorForwardVector();
+			FVector SpawnPos = OwnerChar->GetActorLocation() + FowardVec * 100;
+
+			FRotator CharacterRotation = FowardVec.Rotation();
+
+			FVector BoneDirection = FowardVec + FVector(0.0, 0.0, 0.3);
```

Comment:
Can this be a UProperty so we can change it in Unreal, so we can adjust how far the character throws it easier?

Reply:
I will change that when we have the first person perspective, right now it is like this just for me to test the trigger